app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java

```diff
@@ -190,6 +194,75 @@ public static boolean almostEquals( TransmitterRawData e1, TransmitterRawData e2
     	return mt.ReadFromMongo(numberOfRecords);
     }
 
+    private static OkHttpClient httpClient = null;
+
+
+    // read from http source like cloud hosted parakeet receiver.cgi / json.get
+    public static List<TransmitterRawData> ReadHttpJson(String url, int numberOfRecords) {
+        List<TransmitterRawData> trd_list = new LinkedList<TransmitterRawData>();
+
+        try {
+
+            if (httpClient == null) {
+                httpClient = new OkHttpClient();
+                // suitable for GPRS
+                httpClient.setConnectTimeout(30, TimeUnit.SECONDS);
+                httpClient.setReadTimeout(60, TimeUnit.SECONDS);
+                httpClient.setWriteTimeout(20, TimeUnit.SECONDS);
+            }
+
+            Gson gson = new GsonBuilder().create();
+
+            // simple HTTP GET request
+            // n=numberOfRecords for backfilling
+            // r=sequence number to avoid any cache
+            // expecting json reply like the standard json server in dexterity / python pi usb / parakeet
+            Request request = new Request.Builder()
+
+                    // Mozilla header facilitates compression
+                    .header("User-Agent", "Mozilla/5.0")
```

Comment:
okhttp defaults to Accept-encoding: gzip but google app engine servers ignore this unless the user agent contains "Mozilla" - This is documented in their faq as a mechanism to avoid certain compression handling bugs.